Objects/unicodeobject.c

```diff
@@ -2875,47 +2875,45 @@ unicode_fromformat_arg(_PyUnicodeWriter *writer,
 static int
 unicode_from_format(_PyUnicodeWriter *writer, const char *format, va_list vargs)
 {
-    writer->min_length += strlen(format) + 100;
+    Py_ssize_t len = strlen(format);
+    writer->min_length += len + 100;
     writer->overallocate = 1;
 
-    va_list vargs2;
-    const char *f;
-
     // Copy varags to be able to pass a reference to a subfunction.
+    va_list vargs2;
     va_copy(vargs2, vargs);
 
-    for (f = format; *f; ) {
+    int is_ascii = (ucs1lib_find_max_char((Py_UCS1*)format, (Py_UCS1*)format + len) < 128);
```

Comment:
Add a comment that this is needed because we use `_PyUnicodeWriter_WriteASCIIString`.

Reply:
Good idea, done.